pallets/gear/src/internal.rs

```diff
@@ -591,22 +590,18 @@ where
                 Self::remove_gas_reservation_with_task(dispatch.source(), reservation_id);
             }
 
-            // Calculating correct hold bound to lock gas.
-            let maximal_hold = hold_builder.maximum_for_message(dispatch.id());
```

Comment:
did I get it right, that it was a copy paste from `wait_dispatch` and it's incorrect calculating `maximal_hold` here?

Reply:
Yeah, no idea why here we should take maximum for message if we have pre-defined amount of blocks for storing in dispatch stash. Probably copy-pasted from wait_dispatch where wait(None) does it for maximum amount of blocks